07-routing-gossip.md

```diff
@@ -798,6 +897,42 @@ and allows bootstrapping for new nodes as well as updating for nodes that
 have been offline for some time.  The `gossip_queries` option
 allows for more refined synchronization.
 
+### Inventory-based Rebroadcasting
+
+With the simple rebroadcasting scheme described above, nodes that are connected to many different peers will receive duplicates for the same gossip messages.
+
+A more efficient scheme based on inventory messages, similar to how transactions and blocks are propagated on the bitcoin p2p network, can optionally be used: if the `option_inv_gossip` feature bit is set, instead of broadcasting `channel_annoucement` and `channel_update` messages, nodes will broadcast a `gossip_inventory` message which is similar to a `reply_channel_range` message and can be used to filter announcements by timestamp or by content (checksum):
+
+1. type: 266 (`gossip_inventory`)
+2. data:
+    * [`32`:`chain_hash`]
+	* [`tlvs`:`gossip_inventory_tlvs`]
+
+1. tlvs: `gossip_inventory_tlvs`
+2. types:
+   1. type: 1 (`channel_inv`)
+   2. data:
+      * [`tlv_len`:`encoded_short_ids`]
+   1. type: 3 (`timestamps`)
+   2. data:
+	  * [`1`:`encoding_type`]
+	  * [`tlv_len-1`:`encoded_timestamps`]
+   1. type: 5 (`checksums`)
+   2. data:
+        * [`tlv_len`:`encoded_checksums`]
+   1. type: 7 (`node_inv`)
+   2. data:
+      * [`tlv_len`:`node_ids`]
+ 
+`channel_inv` contains information about channel announcements and updates. It uses the exact same encoding as `reply_channel_range` and includes:
+ - a list of `short_channel_id`s
+ - an optional `timestamps`, which includes `channel_update` timestamps
+ - an optional `checksums`, which includes `channel_update` checksums
+
+ `node_inv` contains a list of node ids, 33 bytes for each node id.
```

Comment:
i imagined this would also include a timestamp, otherwise we may request redundant node announcements

Reply:
Yes, see comment above.